### PR TITLE
Update build-tools versions to latest (0.44.0)

### PR DESCRIPTION
**Pull request by ChumpChief**
This also requires updates to some typetests since the format has changed.

~~One in particular to note:  `@fluidframework/server-routerlicious` seems to have always had dubious typetests - it tries to point at a nonexistent index.js/index.d.ts, which apparently didn't blow up for old typetests generation but does for the new one.  Seems that previously it arbitrarily found one of the multiple `create()` functions in the package when generating the typetest file, but obviously not any of the rest of the contents.~~

~~I elected to just disable typetests for that package in this PR since it wasn't obvious that it was ever testing anything correctly.  Based on the package readme it sounds like the intent was for customers to reach into dist/ directly (and that's what I see in practice), but this isn't recommendable.  So the best solution would be to formalize the exports for the package and then reenable typetests against those exports.~~

UPDATE: No changes to routerlicious in this PR due to an issue with the docker build.

### CODE REVIEW

File: server/routerlicious/packages/routerlicious/src/test/types/validateServerRouterliciousPrevious.generated.ts [hunk 1]
```diff
@@ -1,48 +0,0 @@
...
- * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
- * Licensed under the MIT License.
- */
-
-/*
- * THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
- * Generated by fluid-type-test-generator in @fluidframework/build-tools.
- */
-
-import type * as old from "@fluidframework/server-routerlicious-previous";
```
Comment (ChumpChief) [inline server/routerlicious/packages/routerlicious/src/test/types/validateServerRouterliciousPrevious.generated.ts:11]: This has no direct exports -- everything is in dist/.

File: server/routerlicious/packages/routerlicious/src/test/types/validateServerRouterliciousPrevious.generated.ts [hunk 2]
```diff
@@ -1,48 +0,0 @@
...
-import type * as current from "../../index.js";
```
Comment (ChumpChief) [inline server/routerlicious/packages/routerlicious/src/test/types/validateServerRouterliciousPrevious.generated.ts:12]: This file never existed.

File: build-tools/package.json [hunk 3]
```diff
@@ -73,9 +73,9 @@
 		"@commitlint/cli": "^17.6.6",
 		"@commitlint/config-conventional": "^17.6.6",
 		"@commitlint/cz-commitlint": "^17.5.0",
-		"@fluid-tools/build-cli": "~0.40.0",
+		"@fluid-tools/build-cli": "~0.44.0",
```
Comment (tylerbutler) [inline build-tools/package.json:76]: Unfortunately, due to quirks in build-tools you need to also manually update references like this in some build-tools packages:

```json
		"@fluidframework/build-tools-bin": "npm:@fluidframework/build-tools@~0.40.0",
```

**Comment by ChumpChief** [comment]
The docker build is having some trouble finding the root, reverting routerlicious changes for now.

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**